File: java-gdb-activities-20/activity18/src/com/gdb/logging/LogDestination.java
package com.gdb.logging;

import com.gdb.command.TransactionCommand;
import java.util.List;

/**
 * Implementor interface for Bridge Pattern decoupling transaction log storage backends.
 */
public interface LogDestination {
    // ============================================================
    // 📝 STEP 1: Add write(TransactionCommand cmd)
    // ============================================================
    // TODO: declare write method
    void write(TransactionCommand cmd);

    // ============================================================
    // 📝 STEP 2: Add readAll()
    // ============================================================
    // TODO: declare readAll method returning List<TransactionCommand>
    List<TransactionCommand> readAll();

    // ============================================================
    // 📝 STEP 3: Add clear()
    // ============================================================
    // TODO: declare clear method
    void clear();

    // ============================================================
    // 📝 STEP 4: Add getDestinationName()
    // ============================================================
    // TODO: declare getDestinationName method returning String
    String getDestinationName();
}

File: python-gdb-activities-20/activity18/gdb/logging/log_destination.py
# gdb/logging/log_destination.py
from abc import ABC, abstractmethod
from typing import List
from gdb.domain.transaction import Transaction

# ============================================================
# 📝 STEP 1: Define LogDestination Implementor Interface
#
# INSTRUCTIONS:
#   1. abstractmethod write(transaction: Transaction) -> None
#   2. abstractmethod read_all() -> List[Transaction]
#   3. abstractmethod get_destination_name() -> str
#
# HINT: Decorate each method with @abstractmethod; the body can simply be `pass`.
#       MemoryLogDestination (provided) shows a finished implementor of this interface.
# ============================================================
class LogDestination(ABC):
    """Implementor Interface in the Bridge Pattern for logging storage destinations."""

    # TODO: declare the three abstract methods listed in STEP 1
    pass
